### PR TITLE
Fix failing tests on Header-Sidebar feature branch

### PR DESCRIPTION
**Pull request by akash-joshi**
## 🌟 What is the purpose of this PR?

One of the previous commits to the `header-sidebar-update` branch had broken the CI. More specifically, `jest` and `playwright` tests. This PR is responsible for fixing those tests so all future PRs can benefit from working off of a stable branch.

Changes will be described in the comments below.

## 🛡 What tests cover this?

- Existing playwright and jest tests cover the changes made in this PR.

## ❓ How to test this?

Run `yarn test` or `yarn test:playwright` on the `frontend` package locally. Otherwise, rely on the CI output presented by GitHub Actions.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201925844970643

### CODE REVIEW

File: packages/hash/frontend/src/testUtils/testUtils.tsx [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
 const customRender = (
   ui: React.ReactElement,
   options: CustomRenderOptions = {},
 ) => {
   const { mocks = [] } = options;
   const Wrapper: React.FC = ({ children }) => (
-    <MockedProvider mocks={mocks} addTypename={false}>
-      {children}
-    </MockedProvider>
+    <ThemeProvider theme={theme}>
```
Comment (akash-joshi) [inline packages/hash/frontend/src/testUtils/testUtils.tsx:22]: A `ThemeProvider` reusing the existing MUI theme is added to fix the `jest` tests.

File: packages/hash/playwright/tests/entity-creation.spec.ts [hunk 2]
```diff
@@ -44,7 +44,9 @@ test("user can create and update entity", async ({ page }) => {
   );
 
   // Expect the created property to be present
-  await expect(page.locator("text=Property1")).toBeVisible();
+  await expect(
+    page.locator(`text=Property1Property1 >> input[type="text"]`),
```
Comment (akash-joshi) [inline packages/hash/playwright/tests/entity-creation.spec.ts:48]: Makes the property locator more specific.

File: packages/hash/playwright/tests/guest-user.spec.ts [hunk 3]
```diff
@@ -59,5 +59,5 @@ test("guest user navigation to inaccessible pages", async ({ page }) => {
     page.locator("text=This page could not be found."),
   ).toBeVisible();
 
-  await expect(page.locator("button")).toHaveText(["Log In", "Sign Up"]);
+  await expect(page.locator("button")).toHaveText(["Sign In", "Sign Up"]);
```
Comment (akash-joshi) [inline packages/hash/playwright/tests/guest-user.spec.ts:62]: Fixes the expected text on signin buttons.

File: packages/hash/frontend/src/testUtils/testUtils.tsx (hunk 1 above)
Comment (CiaranMn) [reply packages/hash/frontend/src/testUtils/testUtils.tsx:22]: Why not import the `theme` that's already created in `src/theme/index.ts`? It saves having to update it in two places, and guarantees we're testing the theme we're using.

**Review by benwerner01 — APPROVED** [review]
Thanks Akash, looks good to me 👍

File: packages/hash/frontend/src/testUtils/testUtils.tsx (hunk 1 above)
Comment (benwerner01) [reply packages/hash/frontend/src/testUtils/testUtils.tsx:18]: This was addressed in https://github.com/hashintel/hash/pull/378/commits/5fd1f743cdb251628c80e685824f16db3bd52407

**Pull request merged**